File: front/src/client/TelegramClient.js
import BaseClient from "src/client/BaseClient";

class TelegramClient extends BaseClient {
    constructor(client) {
        super();
        this.client = client;
        this.base_url = `${this.client.base_url}/telegram`;
    }

    async generatePayload(data) {
        try {
            const response = await this.client.client.post(`${this.base_url}/generate_payload`, data);

            return response.data;
        } catch (error) {
            console.error('Error generating payload:', error);
            throw error;
        }
    }
}

export default TelegramClient

File: src/client/TelegramClient.py
from src.api.messangers.dtos.telegram_dto import TelegramPayloadDTO
from src.client.BaseClient import BaseClient


class TelegramClient(BaseClient):
    def __init__(self, client):
        self.client = client
        self.session = client.session
        self.base_url = f"{self.client.base_url}/telegram"

    async def generate_payload(self, data: TelegramPayloadDTO, *, return_response: bool = False) -> dict:
        response = await self.session.post(f"{self.base_url}/generate_payload", json=data.dict())
        if return_response:
            return response
        return response.json()
